brack-parser/src/curly.rs

```diff
@@ -1,28 +1,25 @@
 use anyhow::Result;
 use brack_tokenizer::tokens::{mock_location, Token};
 
-use crate::{
-    ast::new_curly, error::ParserError, parser::Parser, surrounded, utils::consume_by_kind,
-};
+use crate::error::{DocumentError, ParseTerminationError, ParserError};
+use crate::{ast::new_curly, parser::Parser, surrounded, utils::consume_by_kind};
 
 // "{" ident (expr ("," expr)*)? "}"
 pub fn parse(tokens: &Vec<Token>) -> Result<Parser, ParserError> {
     let (mut consumed, mut new_tokens) =
         consume_by_kind(&tokens, Token::CurlyBracketOpen(mock_location()));
     if !consumed {
-        return Err(ParserError::new_parse_termination_error(
-            "Curly Brackets is not opened.".to_string(),
-            tokens.first().unwrap().clone(),
-        ));
+        return Err(ParseTerminationError::CurlyNotOpened(
+            new_tokens.first().unwrap().get_location(),
```

Comment:
Ideally you should not use `unwrap` and should branch to a state of type `Result`. It can cause panic in an unexpected state, for example if someone accidentally rewrites `consume_by_kind` to a state that does not return a `token`.

Reply:
This `unwrap` is what I wrote, so I apologise.

Reply:
When raising a `ParserError`, it is necessary to pass a `Location`, but currently, as mentioned in the issue #38 , the AST does not have a `Location`. Therefore, to return a `ParserError`, it is necessary to refer to the token's `Location`, resulting in this implementation. These issues will be addressed once the AST is updated to include `Location`.

Reply:
I understand, thank you.